Objective-C/CBLReplicator.mm

```diff
@@ -460,13 +470,19 @@ - (void) removeChangeListenerWithToken: (id<CBLListenerToken>)token {
         return nil;
     }
     
-    if (!_repl) {
-        CBLLogInfo(Sync, @"Trying to fetch pending documentIds without a c4replicator %@", _repl);
+    C4Error c4err = {};
+    C4Replicator* repl = [self c4repl: &c4err];
+    if (!repl) {
+        NSError* err = nil;
+        convertError(c4err, &err);
+        CBLWarnError(Sync, @"%@: Replicator cannot be created: %@", self, err.localizedDescription);
+        if (error)
+            *error = err;
         return nil;
     }
     
-    C4Error c4err = {};
-    C4SliceResult result = c4repl_getPendingDocIDs(_repl, &c4err);
+    c4err = {};
+    C4SliceResult result = c4repl_getPendingDocIDs(repl, &c4err);
```

Comment:
Should we call this in CBL_LOCK(self) to avoid race condition if repl is freed when stopped or offline?